File: tree_primer/tree_primer__ovar_ternary_tree_child.cs
// 実行時間 20ms
using System.Collections.Generic;
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] nkr = ReadLine().Split().Select( int.Parse ).ToArray();
        var d = new Dictionary<int, List<int>>( nkr[0] );

        foreach( int _ in Range( 0, nkr[0] - 1 ) ) {
            int[] a = ReadLine().Split().Select( int.Parse ).ToArray();

            if( d.ContainsKey( a[0] ) != true ) {
                d[a[0]] = new List<int>( nkr[0] ) { a[1] };
            } else {
                d[a[0]].Add( a[1] );
            }
        }
        foreach( int _ in Range( 0, nkr[1] ) ) {
            int[] v = ReadLine().Split().Select( int.Parse ).ToArray();

            WriteLine( d[v[0]][v[1] - 1] );
        }
        return;
    }
}

File: tree_primer/tree_primer__ovar_ternary_tree_child.py
# 実行時間 10ms
import sys

rl = sys.stdin.readline
n, k, r = list(map(int, input().split()))
d = {}

for _ in [0] * (n - 1):
    a, b = list(map(int, input().split()))

    if a not in d:
        d[a] = [b]
    else:
        d[a].append(b)
for _ in [0] * k:
    v, l = list(map(int, input().split()))

    print(d[v][l - 1])
